benchmarks/nel/project.yml

```diff
@@ -36,96 +47,94 @@ workflows:
+    outputs:
+      - "assets/wiki/wiki.sqlite3"
 
   - name: create_kb
-    help: "Create the knowledge base and write it to file"
+    help: "Create the knowledge base and write it to file."
     script:
-      - "python ./scripts/create_kb.py reddit ${vars.vectors_model}"
+      - "python ./scripts/create_kb.py ${vars.dataset} ${vars.vectors_model}"
     deps:
-      - "assets/reddit"
+      - "assets/${vars.dataset}"
+      - "assets/wiki/wiki.sqlite3"
     outputs:
-      - "assets/reddit/entities.pkl"
-      - "assets/reddit/entities_failed_lookup.pkl"
-      - "assets/reddit/annotations.pkl"
+      - "assets/${vars.dataset}/entities.pkl"
+      - "assets/${vars.dataset}/entities_failed_lookup.pkl"
+      - "assets/${vars.dataset}/annotations.pkl"
     outputs_no_cache:
-      - "temp/reddit/kb"
-      - "temp/reddit/nlp"
+      - "temp/${vars.dataset}/kb"
+      - "temp/${vars.dataset}/nlp"
 
   - name: compile_corpora
-    help: "Compile corpora, separated in in train/dev/test sets"
+    help: "Compile corpora, separated in in train/dev/test sets."
     script:
-      - "python ./scripts/compile_corpora.py reddit"
+      - "python ./scripts/compile_corpora.py ${vars.dataset} True"
     deps:
-      - "assets/reddit/entities.pkl"
-      - "assets/reddit/entities_failed_lookups.pkl"
-      - "assets/reddit/annotations.pkl"
-      - "temp/reddit/kb"
-      - "temp/reddit/nlp"
+      - "assets/${vars.dataset}/entities.pkl"
+      - "assets/${vars.dataset}/entities_failed_lookups.pkl"
+      - "assets/${vars.dataset}/annotations.pkl"
+      - "temp/${vars.dataset}/kb"
+      - "temp/${vars.dataset}/nlp"
       - "configs/datasets.yml"
     outputs_no_cache:
-      - "corpora/reddit/train.spacy"
-      - "corpora/reddit/dev.spacy"
-      - "corpora/reddit/test.spacy"
+      - "corpora/${vars.dataset}/train.spacy"
+      - "corpora/${vars.dataset}/dev.spacy"
+      - "corpora/${vars.dataset}/test.spacy"
 
   - name: train
-    help: "Train a new Entity Linking component. Pass --gpu_id GPU_ID to train with GPU"
+    help: "Train a new Entity Linking component. Pass --vars.gpu_id GPU_ID to train with GPU. Training with some datasets may take a long time!"
     script:
-      - "python -m spacy train configs/${vars.config} \
-          --paths.dataset_name reddit \
-          --output training/reddit \
-          --paths.train corpora/reddit/train.spacy \
-          --paths.dev corpora/reddit/dev.spacy \
-          --paths.kb temp/reddit/kb \
-          --paths.base_nlp temp/reddit/nlp \
-          -c scripts/custom_functions.py"
+      - "bash scripts/train.sh ${vars.dataset} ${vars.config} ${vars.training_max_steps} ${vars.gpu_id} "
     deps:
-      - "temp/reddit/kb"
-      - "temp/reddit/nlp"
-      - "corpora/reddit/train.spacy"
-      - "corpora/reddit/dev.spacy"
+      - "temp/${vars.dataset}/kb"
+      - "temp/${vars.dataset}/nlp"
+      - "corpora/${vars.dataset}/train.spacy"
+      - "corpora/${vars.dataset}/dev.spacy"
```

Comment:
The `outputs` specification is missing for the `train` command

Reply:
Added. Btw, is there a good reason to use `output_no_cache` for anything we produce here? Just noticed that I'm using `output_no_cache` in most steps, but don't remember why :upside_down_face:

Reply:
cf https://spacy.io/usage/projects#deps-outputs :-)